### DISCUSSION TITLE
Options to improve solving of quantified formulas with arrays

### DISCUSSION THREAD

**Discussion by baierd** [cvc5/cvc5#8594] [Q&A] [answered]
Hello CVC5 team,

is it possible to improve solving of quantified formulas with arrays? Most of the tests (that other SMT solvers can solve, including CVC4) fail with CVC5. We use the Java API of CVC5.

Example:
`(forall x in [10..20] . b[x] = 1) AND (exits x in [15..17] . b[x] = 0) ` should be UNSAT but returns SatUnknown

I can provide more examples, including Java API if you need it.

We use the options:
```
    solver.setOption("produce-models", "true");
    solver.setOption("finite-model-find", "true");
    solver.setOption("sets-ext", "true");
    solver.setOption("output-language", "smtlib2");
    solver.setOption("strings-exp", "true");
    solver.setOption("produce-interpols", "all");
```

Best,
Daniel

**Comment by ajreynol** [comment, marked as answer]
A few other options to try:
`--enum-inst`, which enables enumerative instantiation (https://homepage.divms.uiowa.edu/~ajreynol/tacas18.pdf)
`--sygus-inst`, which enables syntax-guided instantiations (https://homepage.divms.uiowa.edu/~ajreynol/tacas21.pdf)

These methods make cvc5 try harder to find instantiations where other techniques may be incomplete.
